### DISCUSSION TITLE
RBAC disabled with Azure AD

### DISCUSSION THREAD

**Discussion by sideeo** [provectus/kafka-ui#3910] [Q&A]
### Issue submitter TODO list

- [X] I've looked up my issue in [FAQ](https://docs.kafka-ui.provectus.io/faq/common-problems)
- [X] I've searched for an already existing issues [here](https://github.com/provectus/kafka-ui/issues)
- [X] I've tried running `master`-labeled docker image and the issue still persists there
- [X] I'm running a supported version of the application which is listed [here](https://github.com/provectus/kafka-ui/blob/master/SECURITY.md)

### Describe the bug (actual behavior)

Good evening. I followed [this](https://github.com/provectus/kafka-ui/discussions/3814#discussioncomment-5936440) article to set up my configuration file (Azure AD) which you can see below, but it doesn't work.

### Your installation details

![ad](https://github.com/provectus/kafka-ui/assets/115739709/1accacb9-1b59-4744-b2e4-ed0312c3a26b)
![rls](https://github.com/provectus/kafka-ui/assets/115739709/691caa23-8218-453b-ad27-6125d093ce46)

### Steps to reproduce

I'm getting back the roles in a token from AAD.
![MicrosoftTeams-image (3)](https://github.com/provectus/kafka-ui/assets/115739709/f12e7752-9f62-4b73-a0a9-3d3b840d526d)

### Logs

I've enabled debugging and found that RBAC is not enabled.

![MicrosoftTeams-image (1)](https://github.com/provectus/kafka-ui/assets/115739709/95816525-ca2d-4aad-a1ef-dc06dc7dd480)

### Additional context

After decoding the token on jwt, I've seen that the roles are available in **id_token**, not in a **access_token**. [This](https://stackoverflow.com/questions/66348270/azure-ad-v2-roles-not-included-in-access-token) article provides more about it.

Could that cause the problem?

**Comment by Haarolean** [comment]
Hi, these logs have nothing to do with RBAC.

Try the following:
1. Adding a config property: `LOGGING_LEVEL_COM.PROVECTUS.KAFKA.UI.SERVICE.RBAC.EXTRACTOR: TRACE`, check the logs after a restart.
2. Check the output of the`/api/authorization` endpoint (your browser dev console)

There are confirmed cases for Azure AD w/ RBAC on discord so it's a configuration issue.

**Comment by lokeshboggarapu123** [comment]
how to enable rbac?
I was getting this when i tried/api/authorization
{"rbacEnabled":false,"userInfo":null}
